### PR TITLE
fix(client): clean up styles on controls above multi-file editor

### PR DESCRIPTION
**Pull request by bbsmooth**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

This originally started out as just fixing up the keyboard focus outlines for the buttons and tabs control above the multi-file editor, but then I included a few more fixes for those controls while I was at it. I think most of these updates are not controversial, but of course if something doesn't fit then I'll be happy to remove it or make any necessary changes. The main fixes are:

-  Ensure all four sides of keyboard focus outline are showing for all buttons (some of them were only showing three sides)
- Add borders to tab control in light mode so it mimics look of tab control in dark mode (I think this is necessary for accessibility because the shade of gray we are using for the active tab does not have enough contrast).
- Add separator between adjacent file editor buttons when both are expanded (does not work in Firefox at the moment but I think that's OK since this is not an essential fix).
- Fix hover state appearance on toggle buttons so that it reflects a state change has happened as soon as the button is clicked.

The last one has slightly annoyed me for a while now :slightly_smiling_face: I think it is better if the visual state of the button changes immediately when clicked instead of only after the mouse has moved off of it. The side effect is that the contrast of the hover state is not quite as big now, especially for the darker color. Hover states are not required in WCAG and thus do not have minimum contrast requirements, but we might want to consider creating a new hover background color for the darker button if we feel it does not have enough contrast. Another option would be to do away with the background change on hover and consider other styling changes for the hover state.

### CODE REVIEW

File: client/src/templates/Challenges/classic/classic.css
Comment (Sboonny) [inline client/src/templates/Challenges/classic/classic.css:32]: The new styles are better, just don't find the need for light palete selector, and the selector will get in the way when adding new theme.


```suggestion
.action-row [aria-expanded='true'] {
  background-color: var(--secondary-color);
}

.monaco-editor-tabs [aria-expanded='true'] {
```

the background can be removed from this selector for clarity

```css
.monaco-editor-tabs [aria-expanded='true'],
.action-row [aria-expanded='true'] {
  border-color: var(--secondary-color);
  color: var(--secondary-background);
}
```

File: client/src/templates/Challenges/classic/classic.css
Comment (Sboonny) [inline client/src/templates/Challenges/classic/classic.css:88]:
```suggestion
.monaco-editor-tabs button:not([aria-expanded='true']):hover {
  background-color: var(--primary-background);
}
```

File: client/src/templates/Challenges/classic/editor-tabs.tsx
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:60]: Funny enough, when you have added the operator made me sure that if we separate them indefinitely through a margin, is a better look than the current one we have.

This will remove the need for this logic and the surrounding CSS, but I will leave this to you and @ahmadabdolsaheb to decided.

Comment (bbsmooth) [reply client/src/templates/Challenges/classic/editor-tabs.tsx:60]: @Sboonny Definitely, adding a little space between each button would be a good solution as well, such as I did with the new toggle buttons on the Settings page. In fact, this is what I was originally going to do but I didn't want to make that decision on my own. If you and @ahmadabdolsaheb are OK with a permanent small margin between the buttons then I'll change it.

**Comment by bbsmooth** [comment]
I just realized that we had separate rulesets for the editor tab buttons and the panel display buttons but they were doing the same thing. So I have consolidated the styling for all of these buttons under `.action-row`.

**Comment by ahmadabdolsaheb** [comment]
Thanks for tagging me.

Due to established navigation patterns in mobile UIs, we initially wanted have an IOS's 'segmented control' looking tabs where taping one would remove the previously selected editor from view and  only display the newly selected editor.

Since that is not the current behavior, we can completely separate the buttons all together for now.
The tab separator looks fine. Since the separator adjacent side of the buttons do not have a border, I  argue for a more prominent isolation. Styling the buttons similar to the buttons in tool-panel-group should make the UI more consistant. 

For the mobile only tabs, we don't really need hover states; however, changing the hover styles to active/selected styles should be the simplest way to go.

Also, the selected editor button/tab should be obvious. As a result, it should be styled accordingly.

File: client/src/templates/Challenges/classic/editor-tabs.tsx
Comment (bbsmooth) [reply client/src/templates/Challenges/classic/editor-tabs.tsx]: Added permanent spacing between buttons.

**Comment by bbsmooth** [comment]
@ahmadabdolsaheb I think I got all of your suggestions implemented but let me know if I missed anything or didn't quite get it right.

File: client/src/templates/Challenges/classic/editor-tabs.tsx
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:51]: Although outside the scope of the PR, can you add this as translatable string? 🥺 

Editor is translated on other language `reset-warn`, so to stay consistent with the translation.

File: client/src/templates/Challenges/classic/editor-tabs.tsx
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor-tabs.tsx:50]: Just wondering, what adding `span` do? I am missing what it helps with. 😞 
```suggestion
              <span>{`${challengeFile.name}.${challengeFile.ext}`}</span>
```

File: client/src/templates/Challenges/classic/classic.css
Comment (Sboonny) [inline client/src/templates/Challenges/classic/classic.css:76]: editor-tab-sperator can go 👍 
```suggestion
```

**Review by Sboonny — APPROVED** [review]
This is good to me, this can be merged after the nitpicks 👍.

Thank you for taking care of this, Bruce. it always annoyed me, as well.

File: client/src/templates/Challenges/classic/classic.css
Comment (bbsmooth) [reply client/src/templates/Challenges/classic/classic.css:76]: Good catch, I missed this one.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
